README.md

```diff
@@ -222,37 +225,83 @@ $ nf-core download methylseq --singularity
 
 
 INFO: Saving methylseq
- Pipeline release: 1.0
+ Pipeline release: 1.4
  Pull singularity containers: Yes
- Output directory: nf-core-methylseq-1.0
+ Output file: nf-core-methylseq-1.4.tar.gz
 
 INFO: Downloading workflow files from GitHub
 
+INFO: Downloading centralised configs from GitHub
+
 INFO: Downloading 1 singularity container
-nf-core-methylseq-1.0.simg [762.28MB]  [####################################]  780573/780572
+
+INFO: Building singularity image from dockerhub: docker://nfcore/methylseq:1.4
+INFO:    Converting OCI blobs to SIF format
+INFO:    Starting build...
+Getting image source signatures
+....
+INFO:    Creating SIF file...
+INFO:    Build complete: /domus/h1/phil/GitHub/nf-core/tools/testing/nf-core-methylseq-1.4/singularity-images/nf-core-methylseq-1.4.simg
+
+INFO: Compressing download..
+
+INFO: Command to extract files: tar -xzf nf-core-methylseq-1.4.tar.gz
+
+INFO: MD5 checksum for nf-core-methylseq-1.4.tar.gz: f5c2b035619967bb227230bc3ec986c5
 ```
 
-```console
-$ tree -L 2 nf-core-methylseq-1.0/
+The tool automatically compresses all of the resulting file in to a `.tar.gz` archive.
+You can choose other formats (`.tar.bz2`, `zip`) or to not compress (`none`) with the `-c`/`--compress` flag.
+The console output provides the command you need to extract the files.
 
-nf-core-methylseq-1.0/
+Once uncompressed, you will see the following file structure for the downloaded pipeline:
+
+```console
+$ tree -L 2 nf-core-methylseq-1.4/
+
+nf-core-methylseq-1.4
+├── configs
+│   ├── bin
+│   ├── conf
+│   ├── configtest.nf
+│   ├── docs
+│   ├── LICENSE
+│   ├── nextflow.config
+│   ├── nfcore_custom.config
+│   └── README.md
 ├── singularity-images
-│   └── nf-core-methylseq-1.0.simg
+│   └── nf-core-methylseq-1.4.simg
```

Comment:
Are we downloading `simg` files or `img` files?

Reply:
We're calling the file `.simg`:
https://github.com/nf-core/tools/blob/2d324b4334658e1cd907289b4a7497abf5ef4191/nf_core/download.py#L268

Is this wrong? I forget the difference.

We're also doing `pull`, I can never remember if that's better than `build` or not or the same..

https://github.com/nf-core/tools/blob/2d324b4334658e1cd907289b4a7497abf5ef4191/nf_core/download.py#L271

Reply:
This is what I do in sarek for the extra containers:
https://github.com/nf-core/sarek/blob/4bdff8eed3e4bf1a4393b226d2e87f09c2346d35/scripts/download_image.sh#L69
I'll check if it should be `build` or `pull` and `simg` or `img`

Reply:
OK, so images should have the `img` extension.
This is what happen when I do `nextflow run . -profile singularity,test`:

```bash
...
Pulling Singularity image docker://nfcore/sarek:dev [cache /home/maxime/workspace/nf-core_sarek/work/singularity/nfcore-sarek-dev.img]
```
But according to singularity docs, correct extension should be `sif`, but their docs mention `img` and `simg` as well..

Also, I do think `build` is the correct way to do what we're doing cf https://sylabs.io/guides/3.5/user-guide/quick_start.html#download-pre-built-images

But I think that Nextflow is using pull: https://github.com/nextflow-io/nextflow/blob/036af4a6c378e4db442c667691bf8b070e890890/modules/nextflow/src/test/groovy/nextflow/container/SingularityCacheTest.groovy#L96

Reply:
`img` (1.X-2.3X I _think_) and `simg` (2.4- <3.0X) were the legacy names prior to Singularity 3.X versions that introduced the `sif` format.

Reply:
So... `.sif` then I guess?

Reply:
I think Nextflow however always names them IMG, no matter which Singularity version is used... 

https://github.com/nextflow-io/nextflow/blob/036af4a6c378e4db442c667691bf8b070e890890/modules/nextflow/src/test/groovy/nextflow/container/SingularityCacheTest.groovy#L128

Reply:
Good point.
Should we make a PR on Nextlfow about that?
Or do we really care that much about the extension?

Reply:
The problem we had previously was some people used Singularity V2.x but a Singularity cache with images for 3.X - that caused some conflicts as these are not interchangeable. Since we all migrated to 3.4.X now, we're fine since then 👍 

Not sure this is worth the effort tbh :-)

Reply:
I agree